actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -198,10 +198,12 @@ def fragment_cache_key(key)
 
     private
 
-      def fragment_name_with_digest(name) #:nodoc:
-        if @virtual_path
+      def fragment_name_with_digest(name, virtual_path: nil) #:nodoc:
+        virtual_path ||= @virtual_path
+        if virtual_path
           names  = Array(name.is_a?(Hash) ? controller.url_for(name).split("://").last : name)
-          digest = Digestor.digest name: @virtual_path, finder: lookup_context, dependencies: view_cache_dependencies
+          dependencies = defined?(view_cache_dependencies) ? view_cache_dependencies : []
```

Comment:
Why are we checking `defined?` here? Do we have to pass an empty array or will nil do? If the latter, we could do:

```ruby
dependencies = view_cache_dependencies if defined?(view_cache_dependencies)
```